### ISSUE TITLE
Custom block broken since 8.4.0

### ISSUE THREAD

**Issue by perfectpeachstudio** [WordPress/gutenberg#23627]
Since updating to Gutenberg 8.4.0, my custom block doesn't work anymore when the Gutenberg plugin is active but still works in the builtin WordPress block editor.

Everything works fine in Gutenberg 8.3.0.
As soon as I update to version 8.4.0, the block doesn't show up anymore in the list of the available blocks when I try to add a new one and all the blocks already on the page fail to load with the following error:

> Your site doesn’t include support for the "waveplayer/block" block. You can leave this block intact or remove it entirely.

No errors are triggered in the PHP debug log or the Javascript browser console.
Any help troubleshooting this issue is greatly appreciated. Honestly, I wouldn't know where to start.
I have already checked the changelog and haven't found any info about a possible break of blocks perfectly working in the previous version.

Please advise.

**Comment by oxyc** [comment]
If you can join WordPress [slack](https://make.wordpress.org/chat/), send me a message and maybe I can help debug it. My username is `oxy`.

**Comment by perfectpeachstudio** [comment]
Thanks @oxyc, I believe I found the issue.
Apparently, until version 8.3.0 (and in WordPress without Gutenberg installed), **REGISTERING** the block editor assets is enough to activate the block in the editor.

Starting from version 8.4.0, for some reasons that I have not been able to track down yet, simply registering the script is not enough anymore and **ENQUEUING** the block editor assets becomes necessary.

I am not sure why until version 8.3.0 and in WordPress without the Gutenberg plugin installed the mere registration of the assets is enough to activate the block, while in the latest version 8.4.0 this doesn't work anymore. Unless I missed something, this change has not been documented anywhere.

**Comment by oxyc** [comment]
Ah, it might have to do with https://github.com/WordPress/gutenberg/pull/22754

**Comment by perfectpeachstudio** [comment]
Thanks for pointing that out.

The problem was certainly connected to the PR you mentioned and caused by the actions I used to register the block assets.
I used to **register** the assets using the `enqueue_block_assets` and the `enqueue_block_editor_assets` action hooks.
This was working as expected until Gutenberg 8.3.0 and with the current version of WordPress.

With this new approach, though, apparently those actions are too late to register the assets. So, moving the registration of the assets to the `init` action hook solves the issue.

Thanks for your help!
